articles/tokens/idp.md

```diff
@@ -1,49 +1,79 @@
+Third-party access tokens are issued by Identity Providers (including Social providers like Facebook and Google) when a user authenticates with that provider. These access tokens can be used to call the API of the third-party provider that issued them.
+
+## How to obtain Identity Provider access tokens
+
+Identity Provider (IdP) access tokens can be obtained after the user has authenticated with the IdP by making an HTTP GET call to the `/api/v2/user/{user-id}` endpoint containing an Auth0 API token generated with  `read:user_idp_tokens` scope. 
+
+The following steps outline the required procedure.
 
-## How to get Identity Provider access tokens
+### 1. Get the user id from the user profile
 
-Identity Provider access tokens can be obtained in one of two ways.  First, if a user authenticates to a social Identity Provider, such as Facebook, an Identity Provider access token for that social Identity Provider will be returned in the first element of the `identities` array within the user profile object returned by Auth0.
+First, you will need to retrieve the `user_id` property from the user profile after successful authentication.
 
-For example, if authentication is invoked via a call to the Lock widget’s .show method, the following code would be used to put the Identity Provider access token into a variable called `identityProviderAccessToken`. .
+For example, if you are using [Lock](/libraries/lock), you can get the `user_id` from the returned profile with this code:
 
-```js
+```
 lock.show({
   responseType: 'token',
-  authParams: {
-     scope: 'openid name email'
-  }
-}, function(err, profile, token) {
-    identityProviderAccessToken = profile.identities[0].access_token;
+}, 
+  function(err, profile, token) {
+    user-id = profile.user_id;
 });
 ```
 
-More information is available at [User Profile](/user-profile) and [Normalized User Profile](/user-profile/normalized) contains examples of the `identities` array and profiles as returned by various social providers.
+**NOTE:** For examples of profiles as returned by various social providers, see: [Normalized User Profile](/user-profile/normalized) 
+
+### 2. Call the Auth0 API
+
+Next, you will need to call the [GET Users by Id](/api/management/v2#!/Users/get_users_by_id) endpoint with the value of `user-id` obtained in the previous step.
+
+For example, the following code will output the value of the Idp access token to the console:
+
+```
+var request = require("request");
+
+var options = { method: 'GET',
+  url: 'http://{account.namespace}/api/v2/users/{user-id}',
+  headers: { authorization: 'Bearer YOUR_API_TOKEN' } };
+
+request(options, function (error, response, user) {
+  if (error) throw new Error(error);
 
-## How to control contents of Identity Provider access tokens
+  console.log(user.identities[0].access_token);
+});
+```
+**NOTE:** The `YOUR_API_TOKEN` must contain the `read:user_idp_tokens` scope.
```

Comment:
We are not showing how to get that auth0 api token to make the call. is there a way to generate the token programmatically?